### ISSUE TITLE
Notable not start after first init

### ISSUE THREAD

**Issue by tomasdelvechio** [notable/notable#1106]
- **OS Version**: Ubuntu 18.04
- **Notable Version**: 1.8.4 (via snap)

### Current behavior

After install via snap (`sudo snap install notable`), Notable start fine. But if I close the app, and try to launch again, the app isn't start.

### Expected behavior

Re-launch the after install.

### Some details

If I launch notable with strace, I found this last lines

```
...
openat(AT_FDCWD, ".", O_RDONLY|O_NOFOLLOW|O_CLOEXEC|O_PATH|O_DIRECTORY) = 3
fstat(3, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
chdir("/")                              = 0
getresuid([1000], [1000], [1000])       = 0
getresgid([1000], [1000], [1000])       = 0
write(2, "need to run as root or suid", 27need to run as root or suid) = 27
write(2, "\n", 1
)                       = 1
exit_group(1)                           = ?
+++ exited with 1 +++
```

If i try to run with sudo, doesn't work (I get this error):

```
mkdir: cannot create directory '/run/user/0': Permission denied                                                                
```

If I use the deb package, Notable work fine, but I check manually for app upgrades.

**Comment by fabiospampinato** [comment]
The previous snap version published on the store didn't really work, there was an issue with its snap's "strict" confinement. I've just now published a working version that uses the "classic" confinement instead, can you reproduce the issue with that?

**Comment by tomasdelvechio** [comment]
I try to reproduce the issue and this still happens :(

My steps:

- remove apt version: `sudo apt purge notable`
- install snap version: `sudo snap install notable --classic`
- Start Notable the first time: ok
- Close Notable
- Open Notable again: Don't start anymore.

**Comment by fabiospampinato** [comment]
That's pretty weird, I can't reproduce that 🤔

**Comment by tomasdelvechio** [comment]
Ok, if you know some tips for debug data or dump execution traces over snap installation, I would help you.

**Comment by fabiospampinato** [comment]
Unfortunately the app doesn't really ship with any custom debugging capabilities. From the looks of it it seems that the app itself isn't even able to start, perhaps the issue happens at the snap level.

Any idea why the app would try to create the `/run/user/0` directory?

Could you post here the content of the `~/.notable.json` file?

**Comment by tomasdelvechio** [comment]
> Any idea why the app would try to create the `/run/user/0` directory?

No. I'm afraid I don't idea about snap works.
 
> Could you post here the content of the `~/.notable.json` file?

In ubuntu, this file locate in `~/.config/Notable/notable.json` and the content is:

```json
{"width":850,"height":525,"x":215,"y":263,"isMaximized":true,"isFullScreen":false,"displayBounds":{"x":0,"y":0,"width":1280,"height":1024}}
```

**Comment by fabiospampinato** [comment]
That's another thing, does `~/.notable.json` not exist in your computer?

**Comment by tomasdelvechio** [comment]
> That's another thing, does `~/.notable.json` not exist in your computer?

sorry, after re install last time via apt I don't start the app and then  the file not existis yet.

The file `~/.notable.json` contains:

```json
{
        "monaco": {
                "editorOptions": {
                        "minimap": {
                                "enabled": false
                        },
                        "lineNumbers": "off",
                        "scrollBeyondLastLine": false,
                        "wordWrap": "bounded"
                }
        },
        "sorting": {
                "by": "title",
                "type": "ascending"
        },
        "theme": "dark",
        "tutorial": false,
        "cwd": "/local/path"
}
```

**Comment by fabiospampinato** [comment]
There doesn't seem anything wrong with that 🤔 did you ever see the tutorial notes when the app launched the first time? is your data directory at "/local/path"?

**Comment by tomasdelvechio** [comment]
> There doesn't seem anything wrong with that thinking did you ever see the tutorial notes when the app launched the first time? is your data directory at "/local/path"?

Yes, I change the value for the comment, but the directory is fine because:

1. The first time when Notable up via snap, my notes load fine and I can edit whitout problems
2. With snap in `.deb`, I use the directory diary.

**Comment by tetafro** [comment]
Seems more like a platform issue (snap/ubuntu). I had the same issue with Bitwarden app installed via snap. Switched to dep-package and everything's fine now.

**Comment by fabiospampinato** [comment]
@tetafro Did you install the latest snap published in the store recently? There was an issue with the previous one, maybe that was it.

**Comment by tetafro** [comment]
> Did you install the latest snap published in the store recently?

It was `1.8.4-2`, the one which requires `--classic` key

**Comment by StuartMacKay** [comment]
The executable in `/snap/notable/current/notable` runs fine when you call it directly on the command line. It seems that snap is unable to run it for some reason.
